src/adagrams.js

```diff
@@ -1,8 +1,380 @@
+      lettersForUser.push(poolLetters[num]);
+    }
+
+    return lettersForUser;
+  }
+
+  static usesAvailableLetters(input,lettersInHand){
+    const inputUpper = input.toUpperCase();
+    const inputArray = inputUpper.split('');
+
+    const checkLetter = [];
+
+    inputArray.forEach((letter)=>{
+      const letterTrack = [];
+      for(let i in lettersInHand){
+        if(letter === lettersInHand[i]){
+          letterTrack.push(i);
+        }
+      }
+      if (letterTrack.length === 0){
+        return false;
+      }
+      else {
+        const toSplice = letterTrack[0];
+        checkLetter.push(lettersInHand.splice(toSplice,1));
+      }
+    });
+    return checkLetter.length === inputArray.length;
+  }
+  static scoreWord(word){
+    const wordArray = word.toUpperCase().split('');
+    const score = {};
+    // assign score rules
+    ['A','E','I','O','U','L','N','R','S','T'].forEach((char)=>{
+      score[char] = 1;
+    });
+
+    ['D','G'].forEach((char)=>{
+      score[char] = 2;
+    });
+
+    ['B','C','M','P'].forEach((char)=>{
+      score[char] = 3;
+    });
+
+    ['F','H','V','W','Y'].forEach((char)=>{
+      score[char] = 4;
+    });
+
+    ['J','X'].forEach((char)=>{
+      score[char] = 8;
+    });
+
+    ['Q','Z'].forEach((char)=>{
+      score[char] = 10;
+    });
+
+    score['K'] = 5;
+    let totalScore = 0;
+    for(let eachWord of wordArray){
+      for(let [key,value] of Object.entries(score)){
+        if(eachWord === key){
+          totalScore += value;
+        }
+      }
+    }
+    if(wordArray.length >= 7){
+      totalScore +=8;
+    }
+    return totalScore;
+  }
+
+  static highestScoreFrom(words){
+    const winnersScore = [];
+    words.forEach((wordInput)=>{
+      let tempScore = this.scoreWord(wordInput);
+      winnersScore.push({word:wordInput, score:tempScore, wordLength: wordInput.length});
+    })
+    let max = 0; 
+    winnersScore.forEach((wordObj)=>{
+      if(wordObj.score > max){
+        max = wordObj.score;
+      }
+    })
+    const maxArray = winnersScore.filter(obj => obj.score === max);
+    if(maxArray.length === 1){
+      const answer = maxArray[0];
+      delete answer.wordLength;
+      return answer;
+    }
+    else {
+
+      const ArrayTenLength = maxArray.filter(obj => obj.wordLength === 10);
+
+      if(ArrayTenLength.length >= 1){
+        const answer = ArrayTenLength[0];
+        delete answer.wordLength;
+        return answer;
+      }
+
+      else {
+        let min = 10;
+        maxArray.forEach((wordObj)=>{
+          if(wordObj.wordLength < min){
+            min = wordObj.wordLength;
+          }
+        })
+        const minArray = maxArray.filter(obj => obj.wordLength === min);
+        const answer = minArray[0];
+        delete answer.wordLength;
+        return answer;
+      }
+
+    }
+    
+
+  }
+
+};
+
+
+
+// Do not remove this line or your tests will break!
+export default Adagrams;
+
+
+/* 
+// wave 1 to 4 Adagrams as Object
```

Comment:
In the future please clean up commented out code before turning in your assignments.